cmd/ig/main.go

```diff
@@ -79,7 +76,36 @@ func main() {
 	rootCmd.AddCommand(common.NewLoginCmd())
 	rootCmd.AddCommand(common.NewLogoutCmd())
 
+	// evaluate flags early; this will make sure that flags for host are evaluated before
+	// calling host.Init()
+	err := commonutils.ParseEarlyFlags(rootCmd, os.Args[1:])
+	if err != nil {
+		// Analogous to cobra error message
+		fmt.Fprintf(os.Stderr, "Error: %v\n", err)
+		os.Exit(1)
+	}
+
 	if err := rootCmd.Execute(); err != nil {
 		os.Exit(1)
 	}
 }
+
+// bindViper initializes viper and binds command flags with environment variables
+func bindViper(command *cobra.Command, args []string) error {
+	v := viper.New()
+
+	v.SetEnvKeyReplacer(strings.NewReplacer("-", "_"))
+	v.SetEnvPrefix(EnvPrefix)
+	v.AutomaticEnv()
+
+	command.Flags().VisitAll(func(f *pflag.Flag) {
```

Comment:
Just for my understanding, the approach here is to bind only the environment variables (not flags) so we can `v.Get()` them and override the flag value if it wasn't set by flag?

Reply:
Yes.

Reply:
Makes sense, thanks. Can you please rebase the PR so I can trigger a CI run?

Reply:
Done.